archai/common/azureml_helper.py

```diff
@@ -0,0 +1,132 @@
+            min_instances=min_instances,
+            # Nodes in cluster
+            max_instances=max_instances,
+            # How many seconds will the node running after the job termination
+            idle_time_before_scale_down=idle_time_before_scale_down,
+            # Dedicated or LowPriority. The latter is cheaper but there is a chance of job termination
+            tier=tier,
+            **kwargs
+        )
+
+        cpu_cluster = ml_client.compute.begin_create_or_update(cpu_compute).result()
+        print(f"AMLCompute with name {cpu_cluster.name} is created, the compute size is {cpu_cluster.size}")
+
+
+def create_environment_from_file(
+        ml_client: MLClient,
+        custom_env_name: str = "aml-archai",
+        description: str = "Custom environment for Archai",
+        tags: dict = {"archai": "1.0.0"},
+        conda_file: str = "conda.yaml",
+        image: str = "mcr.microsoft.com/azureml/openmpi3.1.2-ubuntu18.04:latest",
+        version: str = "0.1.0",
+        **kwargs) -> Environment:
+
+    archai_job_env = Environment(
+        name=custom_env_name,
+        description=description,
+        tags=tags,
+        conda_file=conda_file,
+        image=image,
+        version=version,
+        **kwargs
+    )
+    archai_job_env = ml_client.environments.create_or_update(archai_job_env)
+
+    print(
+        f"Environment with name {archai_job_env.name} is registered to workspace, the environment version is {archai_job_env.version}")
+
+    return archai_job_env
+
+
+def create_job(
+        experiment_name: str,
+        display_name: str,
+        compute_name: str,
+        environment: str,
+        code: str,
+        cli_command: str,
+        outputs: Dict,
+        **kwargs) -> Command:
+
+    job = command(
+        display_name=display_name,
+        outputs=outputs,
+        code=code,
+        command=cli_command,
+        environment=environment,
+        compute=compute_name,
+        experiment_name=experiment_name,
+        **kwargs
+    )
+
+    return job
+
+
+def create_job_from_file(source: Union[str, Path], **kwargs) -> Job:
+    return load_job(source=source, **kwargs)
+
+
+def run_job(ml_client: MLClient, job: Job) -> Job:
+    return ml_client.create_or_update(job)
+
+
+def stream_job_logs(ml_client: MLClient, job: Job):
+    ml_client.jobs.stream(job.name)
+
+
+# TODO How can we return the path that the output was downloaded to?
+def download_job_output(
+        ml_client: MLClient,
+        job_name: str,
+        output_name: str,
+        download_path: Union[str, Path] = "output") -> None:
+
+    try:
+        target_job = ml_client.jobs.get(job_name)
+    except Exception as e:
+        print(f"{e.error}")
+        return None
+
+    if target_job.status == "Completed":
+        ml_client.jobs.download(name=target_job.name, download_path=Path(download_path), output_name=output_name)
+    else:
+        print(f"Job {target_job.name} is not completed yet")
```

Comment:
Should we docstring every public method?

Reply:
Done.